File: project-euler/0004/0004.js
var max = 999;
var min = 101;
var palindrome = -1;

loopA:
for(var a = max; a >= min; a--) {
    if(a % 10 === 0) {
        continue;
    }

    loopB:
    for(var b = max; b >= min; b--) {
        var number = a * b;
        if(number < palindrome) {
            break;
        }
        if(number % 10 === 0) {
            continue;
        }
        if(isPalindrome(number) && number > palindrome) {
            palindrome = number;
        }
    }
}

console.log(palindrome);

function isPalindrome(number) {
    var str = number.toString();
    var reverse = str.split('').reverse().join('');
    return str === reverse;
}

File: project-euler/0004/0004.pl
#!/usr/bin/env perl

$max = 999;
$min = 101;
$palindrome = -1;

foreach $a (reverse $min .. $max) {

    next if($a % 10 == 0);

    foreach $b (reverse $min .. $max) {
        $number = $a * $b;

        last if($number < $palindrome);
        next if($number % 10 == 0);

        if($number > $palindrome && reverse($number) == $number) {
            $palindrome = $number;
        }
    }
}

print $palindrome;

